src/reliable_text.c

```diff
@@ -0,0 +1,433 @@
+        }
+        else if (input[index] >= 'A' && input[index] <= 'Z')
+        {
+            output[outidx++] = input[index] - 'A' + 20;
+        }
+        else if (input[index] >= 'a' && input[index] <= 'z')
+        {
+            output[outidx++] = toupper(input[index]) - 'A' + 20;
+        }
+    }
+    output[outidx] = 0;
+}
+
+static void convert_ota_string_to_callsign_(const char* input, char* output, int maxLength)
+{
+    int outidx = 0;
+    for (size_t index = 0; index < maxLength; index++)
+    {
+        if (input[index] == 0) break;
+        
+        if (input[index] >= 1 && input[index] <= 9)
+        {
+            output[outidx++] = input[index] + 37;
+        }
+        else if (input[index] >= 10 && input[index] <= 19)
+        {
+            output[outidx++] = input[index] - 10 + '0';
+        }
+        else if (input[index] >= 20 && input[index] <= 46)
+        {
+            output[outidx++] = input[index] - 20 + 'A';
+        }
+    }
+    output[outidx] = 0;
+}
+
+static char calculateCRC8_(char* input, int length)
+{
+    unsigned char generator = 0x1D;
+    unsigned char crc = 0; /* start with 0 so first byte can be 'xored' in */
+
+    while (length > 0)
+    {
+        unsigned char ch = *input++;
+        length--;
+
+        // Ignore 6-bit carriage return and sync characters.
+        if (ch == 63 || ch == 48) continue;
+        
+        crc ^= ch; /* XOR-in the next input byte */
+        
+        for (int i = 0; i < 8; i++)
+        {
+            if ((crc & 0x80) != 0)
+            {
+                crc = (unsigned char)((crc << 1) ^ generator);
+            }
+            else
+            {
+                crc <<= 1;
+            }
+        }
+    }
+
+    return crc;
+}
+
+static int reliable_text_ldpc_decode(reliable_text_impl_t* obj, char* dest)
+{
+    char* src = &obj->inbound_pending_bits[RELIABLE_TEXT_UW_LENGTH_BITS];
+    char deinterleavedBits[LDPC_TOTAL_SIZE_BITS];
+    _Complex float deinterleavedSyms[LDPC_TOTAL_SIZE_BITS / 2];
+    float deinterleavedAmps[LDPC_TOTAL_SIZE_BITS / 2];
+    float incomingData[LDPC_TOTAL_SIZE_BITS];
+    float llr[LDPC_TOTAL_SIZE_BITS];
+    unsigned char output[LDPC_TOTAL_SIZE_BITS];
+    int parityCheckCount = 0;
+    
+    if (obj->bit_index == obj->sym_index * 2)
+    {
+        // Use soft decision for the LDPC decoder.
+        
+        int Npayloadsymsperpacket = LDPC_TOTAL_SIZE_BITS / 2;
+        
+        // Deinterleave symbols
+        gp_deinterleave_comp ((COMP*)deinterleavedSyms, (COMP*)&obj->inbound_pending_syms[RELIABLE_TEXT_UW_LENGTH_BITS/2], Npayloadsymsperpacket);
+        gp_deinterleave_float(deinterleavedAmps, &obj->inbound_pending_amps[RELIABLE_TEXT_UW_LENGTH_BITS/2], Npayloadsymsperpacket);
+        
+        float EsNo = 3.0; // note: constant from freedv_700.c
+        
+        symbols_to_llrs(llr, (COMP*)deinterleavedSyms, deinterleavedAmps,
+                        EsNo, obj->fdv->ofdm->mean_amp, Npayloadsymsperpacket);
+    }
```

Comment:
OK so `mean_amp` is estimated [here](https://github.com/drowe67/codec2/blob/master/src/ofdm.c#L1735), it's a time averaged estimate of the pilot symbol amplitude.  I think this is OK, as the mean amplitude for a HF channel won't vary too quickly (e.g. over the PSK Packet length).